modeler/cayenne-modeler/src/main/java/org/apache/cayenne/modeler/editor/dbimport/DbImportSorter.java

```diff
@@ -0,0 +1,110 @@
+/*****************************************************************
+ *   Licensed to the Apache Software Foundation (ASF) under one
+ *  or more contributor license agreements.  See the NOTICE file
+ *  distributed with this work for additional information
+ *  regarding copyright ownership.  The ASF licenses this file
+ *  to you under the Apache License, Version 2.0 (the
+ *  "License"); you may not use this file except in compliance
+ *  with the License.  You may obtain a copy of the License at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ ****************************************************************/
+
+package org.apache.cayenne.modeler.editor.dbimport;
+
+import org.apache.cayenne.dbsync.reverse.dbimport.Catalog;
+import org.apache.cayenne.dbsync.reverse.dbimport.FilterContainer;
+import org.apache.cayenne.dbsync.reverse.dbimport.IncludeTable;
+import org.apache.cayenne.dbsync.reverse.dbimport.ReverseEngineering;
+import org.apache.cayenne.dbsync.reverse.dbimport.Schema;
+import org.apache.cayenne.dbsync.reverse.dbimport.SchemaContainer;
+import org.apache.cayenne.modeler.dialog.db.load.DbImportTreeNode;
+import org.apache.cayenne.modeler.editor.dbimport.tree.NodeType;
+
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.List;
+
+/**
+ * @since 5.0
+ */
+public class DbImportSorter {
+    private static final Comparator<DbImportTreeNode> NODE_COMPARATOR = Comparator
+            .comparing(DbImportTreeNode::getNodeType)
+            .thenComparing(DbImportTreeNode::getSimpleNodeName);
+
+    public static void sortNode(DbImportTreeNode node) {
+        sortNodeItems(node);
+        syncUserObjectItems(node);
+    }
+
+    public static void sortNodeWithAllChildren(DbImportTreeNode node) {
```

Comment:
A bit confusing naming of these two methods. `sortNode()` seems to sort exactly children, and the other one just goes recursively down. Maybe something like `sortSingleNode(node)` and `sortSubtree(root)` will do better? Not sure as naming is always a problem..

Reply:
You are right. It might be a confusion. I accept your variant
![f073d202f54ddc623e416aa385b711ee](https://user-images.githubusercontent.com/70625960/214821158-4274f72f-1b91-4233-b832-08bcd5149ff0.jpg)